src/gateway/WebSocket.ts

```diff
@@ -21,67 +39,79 @@ export default async function Socket(token: string, wsURL: string, intents: numb
         intents,
         properties: {
           $os: process.platform,
-          $browser: 'DiscordTS',
-          $device: 'DiscordTS',
+          $browser: 'Dis.ts',
+          $device: 'Dis.ts',
         },
       },
     }));
   }
 
-  ws.on('open', () => {
-    identify();
-  });
+  function resume() {
+    if (ws.readyState === 1) {
+      ws.send(JSON.stringify({
+        op: GATEWAY_OPCODES.RESUME,
+        d: {
+          token,
+          session_id,
+          seq,
+        },
+      }));
+    }
+  }
 
-  ws.on('message', (data: Payload) => {
+  function onMessage(data: Payload) {
     data = JSON.parse(String(data));
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const { t, s, op, d } = data;
+    if (s) seq = s;
 
     switch (op) {
       case GATEWAY_OPCODES.DISPATCH:
-        // if (EVENTS[t]) client.emit(EVENTS[t], d);
+      // if (EVENTS[t]) client.emit(EVENTS[t], d);
+        session_id = d.session_id;
         break;
 
       case GATEWAY_OPCODES.HEARTBEAT:
-        break;
-
-      case GATEWAY_OPCODES.IDENTIFY:
-        break;
-
-      case GATEWAY_OPCODES.PRESENCE_UPDATE:
-        break;
-
-      case GATEWAY_OPCODES.VOICE_STATE_UPDATE:
-        break;
-
-      case GATEWAY_OPCODES.RESUME:
+        heartbeat(false);
         break;
 
       case GATEWAY_OPCODES.RECONNECT:
-        break;
-
-      case GATEWAY_OPCODES.REQUEST_GUILD_MEMBERS:
+        clearInterval(interval);
+        ws.removeEventListener('close');
+        ws.terminate();
+        newWS();
+        resume();
         break;
 
       case GATEWAY_OPCODES.INVALID_SESSION:
+        setTimeout(() => {
+          identify();
+        }, 4000);
         break;
```

Comment:
You should really check if the session can be resumed. This can be added afterwards if you aren't adding it now

Reply:
what do you mean exactly

Reply:
See https://discord.com/developers/docs/topics/gateway#resuming